fvm/src/call_manager/default.rs

```diff
@@ -179,6 +186,55 @@ where
         res
     }
 
+    fn validate<K: Kernel<CallManager = Self>>(
+        &mut self,
+        params: crate::kernel::Block, // Message
+        from: ActorID,
+    ) -> Result<InvocationResult> {
+        if self.execution_type != ExecutionType::Validator {
+            return Err(ExecutionError::Fatal(anyhow!(
+                "Tried to call validate with a call manager not configured as a validator."
+            )));
+        }
+        if self.machine.context().tracing {
+            self.trace(ExecutionEvent::Validate {
+                from,
+                params: params.data().to_owned().into(),
+            });
+        }
+        // validate unchecked
+        let result = {
+            // Get the receiver; this will resolve the address.
+            // TODO egg actor creation may be done here in the future ?
+            // how do you validate from a non existing actor? this is a spec TODO
+
+            // Do the actual validate.
+            self.validate_unchecked::<K>(from, params)
+        };
+
+        if self.machine.context().tracing {
+            self.trace(match &result {
+                Ok(InvocationResult::Return(v)) => ExecutionEvent::CallReturn(
+                    v.as_ref()
+                        .map(|blk| RawBytes::from(blk.data().to_vec()))
+                        .unwrap_or_default(),
```

Comment:
Ah, so, is this the logic that would go away if we stopped returning the gas spec?

Reply:
As well as lines 675-710, yes. So the end return value is just to be the exit code the user wants to exit with, _not_ any block handle.